man/systemd.network.xml

```diff
@@ -1468,6 +1468,13 @@ IPv6Token=prefixstable:2002:da8:1::</programlisting></para>
             service type to CS6 (network control) or CS4 (Realtime). Defaults to CS6.</para>
           </listitem>
         </varlistentry>
+        <varlistentry>
+          <term><varname>TCPAdvertisedMaximumSegmentSize=</varname></term>
+          <listitem>
+            <para>Specifies the Path MTU hints given on TCP layer. An unsigned integer in the range 0–4294967294.
+            When unset, the kernel's default will be used.</para>
```

Comment:
this is a size in bytes (network people would probably say "octet", but anyone else just calls it a byte), hence should say so.

Reply:
hmm, the MSS is not the same as the MTU iirc, as the former is without IP headers. see https://en.wikipedia.org/wiki/Maximum_segment_size

hence maybe not claim that this was the path mtu

Reply:
the size 0 is not allowed, is it?

Reply:
Yes it's in bytes. I updated the man.

Reply:
> the size 0 is not allowed, is it?

Size 0 is not rejected, it just means the value is not used. That seems fine to me. It's a way for users to reset it back I guess. Should it be rejeced as invalid instead?

Reply:
TCP segment size can't be zero . We don't actually check for 0 explicitly.  Just not used and will not be applied.

Reply:
That sounds fine to me. Setting green label while waiting for CI, but will wait to see if @poettering is OK with this handling of the zero value before merging.